packages/form-control/src/FormControlMixin.ts

```diff
@@ -0,0 +1,528 @@
+            checked = newChecked;
+
+            /** A requestUpdate call specifically for Lit interactivity */
+            if (this.requestUpdate) {
+              this.requestUpdate();
+            }
+          }
+        });
+      }
+    }
+
+    /** Reset control state when the form is reset */
+    formResetCallback() {
+      this.resetFormControl();
+    }
+
+    /**
+     * A callback for when the controls' form value changes. The value
+     * passed to this function should not be confused with the control's
+     * value property, this is the value that will appear on the form.
+     * In cases where `checked` did not exist on the control's prototype
+     * upon initialization, this value and the value property will be identical;
+     * in cases where `checked` is present upon initialization, this will be
+     * effectively `this.checked && this.value`.
+     */
+    valueChangedCallback(value: any): void {}
+
+    /**
+     * Resets a form control to its initial state
+     */
+    resetFormControl(): void {
+      if (this.hasOwnProperty('checked') && this.checked === true) {
+        this.checked = false;
+      } else {
+        this.value = '';
+      }
+      this.touched = false;
+      this.___forceError = false;
+      this.___checkForError();
+    }
+
+    /**
+     * Check to see if an error should be shown. This method will also
+     * update the internals state object with the --show-error state
+     * if necessary.
+     * @private
+     */
+    ___checkForError(): boolean {
+      if (this.disabled) {
+        return false;
+      }
+
+      const showError = this.___forceError ||
+        (this.touched && !this.validity.valid && !this.focused);
+
+      if (showError) {
+        this.internals.states.add('--show-error');
+      } else {
+        this.internals.states.delete('--show-error');
+      }
+
+      return showError;
+    }
+
+    /**
+     * Set this.touched and this.focused
+     * to true when the element is focused
+     * @private
+     */
+    ___onFocus = (): void => {
+      this.touched = true;
+      this.focused = true;
+      this.___requestUpdate();
+    }
+
+    /**
+     * Reset this.focused on blur
+     * @private
+     */
+    ___onBlur = (): void => {
+      this.focused = false;
+      /**
+       * Set ___forceError to ensure error messages persist until
+       * the value is changed.
+       */
+      if (!this.validity.valid && this.touched) {
+        this.___forceError = true;
+      }
+      this.___requestUpdate();
+    }
+
+    /**
+     * For the show error state on invalid
+     * @private
+     */
+    ___onInvalid = (): void => {
+      this.___forceError = true;
+      this.___requestUpdate();
+    }
+
+    /**
+     * Call Lit's requestUpdate property
+     * @private
+     */
+    ___requestUpdate() {
+      this.___checkForError();
+      /** @ts-ignore */
+      if (this.requestUpdate) {
+        /** @ts-ignore */
+        this.requestUpdate();
+      }
+    }
```

Comment:
What could we do now to prepare the architecture here for integrating with other non-lit libraries?

Reply:
It _might_ not be necessary any more based on how I have it set up. I'll do some testing.

Reply:
OK, this won't be necessary any more. Instead of calling requestUpdate, I'm grabbing the get/set off the prototype or the instance if they exist and calling the after manipulating `value` now, so updates should propagate.